lib/views/question/question_container_view.dart

```diff
@@ -0,0 +1,90 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:go_router/go_router.dart';
+import 'package:kayla_flutter_ic/utils/route_paths.dart';
+import 'package:kayla_flutter_ic/views/question/question_container_state.dart';
+import 'package:kayla_flutter_ic/views/question/question_container_view_model.dart';
+import 'package:kayla_flutter_ic/views/question/question_view.dart';
+
+final questionViewModelProvider = StateNotifierProvider.autoDispose<
+    QuestionContainerViewModel, QuestionContainerState>(
+  (ref) {
+    return QuestionContainerViewModel();
+  },
+);
+
+class QuestionContainerView extends ConsumerStatefulWidget {
+  final String surveyId;
+  final int questionNumber;
+
+  const QuestionContainerView({
+    super.key,
+    required this.surveyId,
+    required this.questionNumber,
+  });
+
+  @override
+  QuestionContainerViewState createState() => QuestionContainerViewState();
+}
+
+class QuestionContainerViewState extends ConsumerState<QuestionContainerView> {
+  @override
+  void initState() {
+    super.initState();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    _setUpData();
```

Comment:
should we call `_setUpData` in `initState` instead? 🤔

Reply:
Question: If we move `_setUpData()` inside `initState()`, the `_setUpData()` will execute the code which changes the state:
```dart
state = QuestionContainerState.success(QuestionContainerUIModel(...));
``` 
But changing state while init must be avoided.
May I use `Future.delayed()` to set up data?

Reply:
@nmint8m We can do it with `Future.delay(0)`

Reply:
@nmint8m it's correct. We can do a trick as @doannimble mentioned 😄 `Future.delayed(Duration(second:0), (){})`